java-checks/src/test/files/checks/UseSwitchExpressionCheck.java

```diff
@@ -0,0 +1,181 @@
+
+class A {
+
+  int field;
+
+  enum DoW {
+    MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY
+  }
+
+  void foo(DoW day) {
```

Comment:
I would add an example with the new switch as well:

```java
    int numLetters;
    switch (day) {  // Noncompliant {{Use expression switch to set value of 'numLetters'.}}
      case MONDAY, FRIDAY, SUNDAY -> {
        numLetters = 6;
      }
      case TUESDAY -> {
        numLetters = 7;
      }
      case THURSDAY, SATURDAY -> {
        numLetters = 8;
      }
      case WEDNESDAY -> {
        numLetters = 9;
      }
      default -> throw new IllegalStateException("Wat: " + day);
    }
```

Reply:
I have not considered the expression switch for the rule before, but good idea! However, this required bigger rework of the rule itself to handle differences in syntax